### DISCUSSION TITLE
GraphQL error: Cannot read property 'updated' of undefined

### DISCUSSION THREAD

**Discussion by lnoel-everlaw** [requarks/wiki#3570] [Error / Bug Report]
**Describe the bug**
When editing a page, I click save and am prompted with: `GraphQL error: Cannot read property 'updated' of undefined`. Then, when closing the page (immediately after hitting save), the `Discard Unsaved Changes?` alert is displayed. I click `Discard unsaved changes`, and yet changes from the editing session are still persisted.

I was able to use the site and edit pages as expected for 2 days before this issue arose.

**To Reproduce**
??? 
1. Edit a page.
2. Click 'save'
3. Click 'close'

**Screenshots**
![Screenshot from 2019-09-30 16-09-52](https://user-images.githubusercontent.com/54292910/65923182-d89bc000-e39c-11e9-9d34-f6546c1aeee1.png)

**Host Info (please complete the following information):**
 - OS: Ubuntu 18.04 running on EC2
 - Wiki.js version: e.g. 2.0.0-beta.303
 - Database engine: postgres 10.10

**Comment by andrefantinato** [comment]
Same error!

**Comment by fireundubh** [comment]
I also get this warning but without the `GraphQL` prefix:
```
[MASTER] warn: Cannot read property 'updated' of undefined
```
Fresh install and 1.x import. Same behavior as reporter described, but the steps are just "edit page, try to save." No need to click close.

I also get this warning:
```
[MASTER] warn: Cannot read property 'created' of undefined
```
That's when creating a new page.

**Comment by fireundubh** [comment]
This issue seems to only occur (for me) when running Wiki.js as a service. When running with `node server start`, Wiki.js doesn't emit these warnings.

**Comment by fireundubh** [comment]
I narrowed this down further.

- When `NODE_ENV` is set to `production`, the issue occurs.
- When `NODE_ENV` is set to `development`, everything works normally.

This suggests that when `NODE_ENV` is set to `development`, some object is being initialized or persisted that needs to be initialized or persisted when `NODE_ENV` is set to `production`.

**Comment by ramirahikkala** [comment]
I get the same error as fireundubh on Linux Mint 18.2 Sonya with wiki.js 2.1.113. 
The error comes while saving the page.

**Comment by jdanderson3** [comment]
I can confirm fireundubh's experience.
Starting the wiki.js with NODE_ENV-development works perfectly,
but when I start with NODE_ENV=production I cannot save any page.  I get  "GraphQL error: Cannot read property 'updated' of undefined" then an infinite box saying it is Rendering.

Centos 7.7
Wiki.js 2.1.114
postgresql-server-9.2.24

**Comment by jdanderson3** [comment]
For me, upgrading to wiki.js 2.2.51 solved this problem.
